src/digraph.rs

```diff
@@ -705,9 +715,8 @@ impl PyDiGraph {
         let mut predec: Vec<&PyObject> = Vec::new();
         let mut used_indices: HashSet<NodeIndex> = HashSet::new();
         for pred in parents {
-            if !used_indices.contains(&pred) {
+            if used_indices.insert(pred) {
```

Comment:
Actually, let's rewrite this to use this API with a `match` statement: https://docs.rs/hashbrown/latest/hashbrown/struct.HashSet.html#method.entry

That way the behavior is clearer. Your code and the existing codes are equivalent, your code saves a lookup but it is not obvious.

Reply:
Changed both predecessors/successors syntax.

Since the usage of this syntax is not consistent across the whole project, I'm going to consolidate it elsewhere.